pallene/checker.lua

```diff
@@ -145,15 +138,22 @@ end
 --
 
 type_check = function(prog_ast)
-    local ok, err = xpcall(check_program, debug.traceback, prog_ast)
+    local co = coroutine.create(function()
+        check_program(prog_ast)
+    end)
+    local ok, err_msg = coroutine.resume(co)
     if ok then
-        return prog_ast, {} -- TODO:  no {}
-    else
-        if getmetatable(err) == TypeError then
-            return false, { err.msg } -- TODO  no {}
+        if coroutine.status(co) == "dead" then
+            -- User's program passed type checker
```

Comment:
The code seems very straight forward, I would remove this comments.

Reply:
I don't think it is *that* straightforward. For example, "err_msg" means different things in each of the branches, and the distinction between type error in the compiler and type error in the user code is inherently confusing. Let's keep the comments :)